test/test_ops.py

```diff
@@ -0,0 +1,122 @@
+    #   throws a runtime error
+    @skipCUDAIfRocm
+    @onlyOnCPUAndCUDA
+    @ops(op_db, unsupported_dtypes_only=True)
+    def test_unsupported_dtypes(self, device, dtype, op):
+        samples = op.sample_inputs(device, dtype)
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        # NOTE: only tests on first sample
+        sample = samples[0]
+        with self.assertRaises(RuntimeError):
+            op(sample.input, *sample.args, **sample.kwargs)
+
+    # Verifies that ops have their supported dtypes
+    #   registered correctly by testing that each claimed supported dtype
+    #   does NOT throw a runtime error
+    @skipCUDAIfRocm
+    @onlyOnCPUAndCUDA
+    @ops(op_db)
+    def test_supported_dtypes(self, device, dtype, op):
+        samples = op.sample_inputs(device, dtype)
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        # NOTE: only tests on first sample
+        sample = samples[0]
+        op(sample.input, *sample.args, **sample.kwargs)
+
+
+class TestGradients(TestCase):
+    exact_dtype = True
+
+    # Tests that gradients are computed correctly
+    @dtypes(torch.double, torch.cdouble)
+    @ops(op_db)
+    def test_grad(self, device, dtype, op):
+        if not op.supports_dtype(dtype, torch.device(device).type):
+            self.skipTest("Skipped!")
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        for sample in samples:
+            with self.subTest(sample=sample):
+                partial_fn = partial(op, **sample.kwargs)
+                self.assertTrue(gradcheck(partial_fn, (sample.input,) + sample.args))
+
+            # NOTE: the following prevents the test from reporting a CUDA memory leak
+            # TODO: extend to multiple input tensors requiring grad
+            del sample.input.grad
+            del sample.input
+
+    # Test that gradients of gradients are computed correctly
+    @dtypes(torch.double, torch.cdouble)
+    @ops(op_db)
+    def test_gradgrad(self, device, dtype, op):
+        if not op.supports_dtype(dtype, torch.device(device).type):
+            self.skipTest("Skipped!")
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        for sample in samples:
+            with self.subTest(sample=sample):
+                partial_fn = partial(op, **sample.kwargs)
+                self.assertTrue(gradgradcheck(partial_fn, (sample.input,) + sample.args,
+                                              gen_non_contig_grad_outputs=False))
+                self.assertTrue(gradgradcheck(partial_fn, (sample.input,) + sample.args,
+                                              gen_non_contig_grad_outputs=True))
+
+                # NOTE: the following prevents the test from reporting a CUDA memory leak
+                # TODO: extend to multiple input tensors requiring grad
+                del sample.input.grad
```

Comment:
TODO: try to reproduce this more precisely and file an issue

Reply:
send snippet to @albanD if can construct

Reply:
I was able to resolve this issue by not using subtest, which must have been holding a reference.